pkg/endpoint/local_endpoint.go

```diff
@@ -115,14 +115,20 @@ func getBackendConfig(nodeObj *v1.Node) (map[string]string, error) {
 
 	//TODO: we should allow the cable drivers to capture and expose BackendConfig settings, instead of doing
 	//      it here.
-	preferredServerStr := os.Getenv("CE_IPSEC_PREFERREDSERVER")
+	preferredServerStr := backendConfig[submv1.PreferredServerConfig]
+
+	if preferredServerStr == "" {
+		preferredServerStr = os.Getenv("CE_IPSEC_PREFERREDSERVER")
+	}
+
 	if preferredServerStr == "" {
 		preferredServerStr = "false"
 	}
 
 	preferredServer, err := strconv.ParseBool(preferredServerStr)
 	if err != nil {
-		return backendConfig, errors.Wrapf(err, "error parsing CE_IPSEC_PREFERREDSERVER bool: %s", preferredServerStr)
+		return backendConfig, errors.Wrapf(err, "error parsing CE_IPSEC_PREFERREDSERVER or node gateway.submariner." +
```

Comment:
pkg/endpoint/local_endpoint.go:130: File is not `gofmt`-ed with `-s` (gofmt)
		return backendConfig, errors.Wrapf(err, "error parsing CE_IPSEC_PREFERREDSERVER or node gateway.submariner." +